packages/react/src/components/Comments/ThreadsSidebar.tsx

```diff
@@ -0,0 +1,95 @@
+import { useBlockNoteEditor } from "../../hooks/useBlockNoteEditor.js";
+import { useUIPluginState } from "../../hooks/useUIPluginState.js";
+import { Thread } from "./Thread.js";
+import { useThreads } from "./useThreads.js";
+import { ThreadData } from "@blocknote/core/comments";
+import { BlockNoteEditor } from "@blocknote/core";
+
+function sortByPosition(
+  editor: BlockNoteEditor<any, any, any>,
+  threads: ThreadData[]
+) {
+  const sortedThreads: ThreadData[] = [];
+
+  editor.prosemirrorState.doc.descendants((node) => {
```

Comment:
The commentsplugin actually does something similar and sets this info in `threadPositions`. It would be cleaner (and more efficient) to expose and reuse that

Reply:
Also, in your current implementation I think you accidentally filter out threads which don't have content anymore in the document (i.e.: when the text has been deleted). Make sure to double check edge-cases and that the `sort` function really only sorts (and does not filter)